www/source/blog/2017-08-09-Habitat-plans-for-net-full-iis-apps.md

```diff
@@ -0,0 +1,301 @@
+}
+```
+
+This bit of powershell script essentially leverages `nuget` and `msbuild` to grab dependencies, compile our code and publish it to our habitat `pkgs` directory which will be archived to a Habitat `hart` package. It should also be noted that in order for Habitat to build this plan, one does NOT need to have Visual Studio installed. Rather you just need a recent framework version of .NET installed which ships by default on server 2012R2 and forward. 
+
+Lets build and package our `hart` for this application:
+
+```
+C:\dev\hab-sln> hab studio enter -w
+   hab-studio: Creating Studio at /hab/studios/dev--hab-sln
+» Importing origin key from standard input
+★ Imported secret origin key core-20170318210306.
+   hab-studio: Entering Studio at /hab/studios/dev--hab-sln
+** The Habitat Supervisor has been started in the background.
+** Use 'hab svc start' and 'hab svc stop' to start and stop services.
+** Use the 'Get-SupervisorLog' command to stream the supervisor log.
+** Use the 'Stop-Supervisor' to terminate the supervisor.
+
+[HAB-STUDIO] Habitat:\src> build .\hab-sln\
+   : Loading C:\hab\studios\dev--hab-sln\src\hab-sln\\habitat\plan.ps1
+   hab-sln: Plan loaded
+   hab-sln: Validating plan metadata
+   ...
+```
+
+This should compile and package our `hab-sln` project. Lets drill in a bit closer on a few of the moving parts of the plan.
+
+### The nuget command line
+
+Note that our plan takes a build dependency on nuget. That is a core habitat plan that packages the `nuget.exe` utility. Our build will use that to do a "package restore" of all packages declared in our projects `packages.config` file. Visual Studio would essentially do the same if we were building inside the IDE.
+
+### Using the right .NET reference assemblies
+
+When you build a .NET project, msbuild searches for reference assemblies matching the framework version you are targeting. You can build a .NET assmbly against any framework version compatible with the CLR version you are using regardless of the version of the .NET framework you have installed. You just need the right reference assemblies. We use the `core/dotnet-47-dev-pack` because we are targetting 4.7 and set the `TargetFrameworkRootPath` to its contents. This may be optional since if the reference assemblies are not present, the build will use the assemblies in your Global Assembly Cache. However, the build could fail if there are APIs used that are not in your locally installed assemblies so it is good practice to reference the appropriate targeting pack.
+
+### Getting the `VisualStudio.Web.targets`
+
+Most of the `MSBUILD` targets, `.prop` files and the `msbuild.exe` needed to build .NET assblies can be found in your local `c:\windows\Microsoft.NET` folder. However the web targets are not. The easiest way to get those without installing Visual Studio is to use `nuget` again and install the `MSBuild.Microsoft.VisualStudio.Web.targets` package and then point `VSToolsPath` to its location.
```

Comment:
As a big fan of the Oxford comma, I'd put a comma after "`.prop` files".

"assblies" => "assemblies"

Need a comma after "However".

"...is to use `nuget` again and install the `MSBuild.Microsoft.VisualStudio.Web.targets` package and then point..." => "... is to use `nuget` to install the `MSBuild.Microsoft.VisualStudio.Web.targets` package and then point..."